src/components/NotificationsList.js

```diff
@@ -0,0 +1,146 @@
+import React, { Component } from 'react';
+import { View, FlatList, RefreshControl } from 'react-native';
+import { Divider, ListItem } from 'react-native-elements';
+import { connect } from 'react-redux';
+import { Spinner, EmptyList, MenuItem, Menu } from './common';
+import {
+  onClientNotificationsRead,
+  onClientNotificationDelete,
+  onCommerceNotificationsRead,
+  onCommerceNotificationDelete
+} from '../actions';
+import { MAIN_COLOR } from '../constants';
+import moment from 'moment';
+import { stringFormatNotificatones } from '../utils';
+
+class NotificationsList extends Component {
+  state = {
+    optionsVisible: false,
+    selectedNotification: null,
+    type: null
+  };
+
+  componentDidMount() {
+    if (this.props.navigation.state.routeName === 'commerceNotificationslist') {
+      this.props.onCommerceNotificationsRead(this.props.commerceId);
```

Comment:
aca estas usando onSnapshot, asi que retorna un listener, entonces acordate que deberias hacer por ejemplo:
`this.unsubscribeNotificationsRead = this.props.onCommerceNotificationsRead(this.props.commerceId);`
y despues en el `componentWillUnmount`, haces:
`this.unsubscribeNotificationsRead && this.unsubscribeNotificationsRead()`
Y como pusiste un refresh, por cada refresh, estas llamando de nuevo al snapshot y creando otro listener, por lo que, al hacer el refresh, deberias hacer tambien el unsubscribe. Y para no repetir tanto codigo, podes sacar todo esto que esta en el componentDidMount, a un metodo aparte que sea por ejemplo `onNotificationsRead` y le agregas arriba de todo esta linea: `this.unsubscribeNotificationsRead && this.unsubscribeNotificationsRead()`. Entonces ese mismo metodo lo podes llamar en el componentDidMount y en el onRefresh. Si no se entendio preguntame jajaj